### PR TITLE
feat: address screening

### PR DESCRIPTION
**Pull request by grothem**
Prevents wallets that are sanctioned or have known links to terrorist financing from using the interface. Leverages the TRM wallet screening functionality to screen the connected account.

### CODE REVIEW

File: src/components/AddressBlocked.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export const AddressBlocked = ({ children }: { children: ReactNode }) => {
+  const { currentAccount } = useWeb3Context();
+  const { isAllowed } = useAddressAllowed();
+
+  if (!isAllowed) {
+    return <AddressBlockedModal address={currentAccount} />;
+  }
+
+  return <>{children}</>;
+};
```
Comment (drewcook) [inline src/components/AddressBlocked.tsx:15]: 🔥 Dope, works for every page

One thing we could do, this adds the navbar to the top:

```jsx
<MainLayout>
  <AddressBlockedModal address={currentAccount} />;
</MainLayout>
```

Comment (grothem) [reply src/components/AddressBlocked.tsx:15]: Nice, good call. Looks better with the nav bar

**Comment by bojanaave** [comment]
All of the screens are blocked on the app.aave.com domain, but I'm still able to connect with classic aave can we block classic as well?

**Comment by MareskoY** [comment]
@grothem need to add disconnect button for blocked modal

**Review by drewcook — APPROVED** [review]
Looks good! 👍

**Review by drewcook — APPROVED** [review]

**Review by defispartan — APPROVED** [review]
👍

**Comment by spartanz51** [comment]
What a shame @grothem . I'm dumping all my aave positions right now

**Comment by spartanz51** [comment]
As explained in the white paper, Aave is decentralized, so this could not be merged. Please close the PR

**Pull request merged**